Makefile

```diff
@@ -17,34 +17,34 @@ quality: ## run quality checks using tox
 	tox -e quality
 
 piptools:
-	pip install -r pip-tools.txt
+	pip install -r requirements/pip-tools.txt
 
 _upgrade: piptools
-	# Generate pins for test requirements.
-	pip-compile test_requirements.in --upgrade -o test_requirements.txt
-
-	# Delete django pin from test_requirements.txt so that tox can control
+	pip-compile requirements/base.in --upgrade -o requirements/base.txt
+	pip-compile requirements/test.in --upgrade -o requirements/test.txt
+	pip-compile requirements/docs.in --upgrade -o requirements/docs.txt
+	# Delete django pin from test.txt so that tox can control
 	# Django version.
-	sed '/^[dD]jango==/d' test_requirements.txt > test_requirements.tmp
-	mv test_requirements.tmp test_requirements.txt
+	sed '/^[dD]jango==/d' requirements/test.txt > requirements/test.tmp
+	mv requirements/test.tmp requirements/test.txt
 
 	# Delete line "-e file:///local/path/to/edx-drf-extensions", which
-	# is a result of the "-e ." hack in test_requirements.in.
-	sed '/^-e /d' test_requirements.txt > test_requirements.tmp
-	mv test_requirements.tmp test_requirements.txt
+	# is a result of the "-e ." hack in test.in.
+	sed '/^-e /d' requirements/test.txt > requirements/test.tmp
+	mv requirements/test.tmp requirements/test.txt
 
 	# Generate pins for pip-tools itself.
-	pip-compile pip-tools.in --upgrade
+	pip-compile requirements/pip-tools.in --upgrade -o requirements/pip-tools.txt
 
 upgrade: ## upgrade test requirement pins
 	CUSTOM_COMPILE_COMMAND="make upgrade" make _upgrade
 
 requirements: piptools ## install test requirements into current env
-	pip-sync test_requirements.txt
-	# `make upgrade` removes Django from test_requirements.txt,
+	pip-sync requirements/test.txt
+	# `make upgrade` removes Django from test.txt,
 	# so we manually install it here.
-	pip install django~=1.11
-	pip install -r docs/requirements.txt
+	pip install -r requirements/base.txt
+	pip install -r requirements/docs.txt
```

Comment:
The `pip-sync` line above uninstalls all the extra dependencies from these files each time `make requirements` is run, then they get re-installed here.  The cleanest solution is probably to use `dev.in` and `dev.txt` files like we have elsewhere to collect all the testing and documentation dependencies, and which doesn't have the Django dependency removed.  Then we can just use `pip-sync` on `requirements/dev.txt`.